tuta-sdk/rust/sdk/src/crypto/rsa.rs

```diff
@@ -32,24 +34,20 @@ impl RSAPublicKey {
 
 const RSA_PUBLIC_EXPONENT: u32 = 65537;
 
-fn public_exponent() -> BigUint {
-    BigUint::new(vec![RSA_PUBLIC_EXPONENT])
-}
-
 impl RSAPublicKey {
     /// Instantiate an RSAPublicKey from a modulus.
     ///
     /// Returns `Err` if the modulus is the wrong size.
-    pub fn from_modulus(modulus: &[u8]) -> Result<Self, RSAKeyError> {
-        rsa::RsaPublicKey::new(BigUint::from_bytes_be(modulus), public_exponent())
+    pub fn from_components(modulus: &[u8], public_exponent: u32) -> Result<Self, RSAKeyError> {
+        rsa::RsaPublicKey::new(BigUint::from_bytes_be(modulus), BigUint::from_bytes_le(&public_exponent.to_le_bytes()))
```

Comment:
https://docs.rs/rsa/latest/rsa/struct.BigUint.html#impl-From%3Cu32%3E-for-BigUint